src/tests/profiler/multiple/multiple.cs

```diff
@@ -10,16 +10,22 @@ namespace Profiler.Tests
 {
     class MultiplyLoaded
     {
-        static readonly Guid MultipleProfilerGuid = new Guid("BFA8EF13-E144-49B9-B95C-FC1C150C7651");
-        static readonly string ProfilerPath = ProfilerTestRunner.GetProfilerPath();
+        private static readonly Guid MultipleProfilerGuid = new Guid("BFA8EF13-E144-49B9-B95C-FC1C150C7651");
+        private static readonly string ProfilerPath = ProfilerTestRunner.GetProfilerPath();
+
+        private static ManualResetEvent _profilerDone = new ManualResetEvent(false);
 
         [DllImport("Profiler")]
         private static extern void PassCallbackToProfiler(ProfilerCallback callback);
 
+        private static void ProfilerDone()
+        {
+            _profilerDone.Set();
+        }
+
         public static int RunTest(String[] args)
         {
-            ManualResetEvent _profilerDone = new ManualResetEvent(false);
-            PassCallbackToProfiler(() => _profilerDone.Set());
+            PassCallbackToProfiler(ProfilerDone);
```

Comment:
I would explicitly cache this delegate given the correctness of the test depends on it. Right now we are relying on the C# compiler to do that caching implicitly. Some versions of the C# compiler (C# 11) do cache it in practice but this is not considered guaranteed behavior by [the spec]( https://github.com/dotnet/docs/pull/28952/files#diff-cbf34ae965b5483118033f6ef2de9734ef3fafce5db37a8fa974e3a6bf7cebe4R128).

Reply:
I will push changes to explicitly cache the delegate because explicit is better, but isn't that link just talking about whether the C# compiler caches and reuses the same delegate in multiple callsites? It could still keep the delegate rooted even if it never uses the same one twice in old versions.